src/main/java/xyz/aadev/generitech/client/gui/power/GuiPowerStorage.java

```diff
@@ -17,25 +17,86 @@
+import xyz.aadev.generitech.client.gui.button.ButtonSides;
+import xyz.aadev.generitech.common.container.power.ContanierPowerStorage;
+import xyz.aadev.generitech.common.network.Network;
+import xyz.aadev.generitech.common.network.messages.power.PacketSides;
 import xyz.aadev.generitech.common.tileentities.power.TileEntityPowerStorage;
 
+import java.io.IOException;
+
 public class GuiPowerStorage extends GuiBase {
 
-    public GuiPowerStorage(InventoryPlayer inventoryPlayer, TileEntityPowerStorage tileEntity) {
-        super(Reference.MOD_ID, new ContanierGenerator(inventoryPlayer, tileEntity));
+    TileEntityPowerStorage tileEntity;
+    private int[] sides;
+
+
+    public GuiPowerStorage(InventoryPlayer inventoryPlayer, TileEntityPowerStorage tileEntity, int[] sides) {
+        super(Reference.MOD_ID, new ContanierPowerStorage(inventoryPlayer, tileEntity));
+        this.sides = sides;
+        this.xSize = 176;
+        this.ySize = 166;
+        this.tileEntity = tileEntity;
     }
 
     @Override
     public void drawBG(int paramInt1, int paramInt2, int paramInt3, int paramInt4) {
-// Do nothing because of not implemented
+        bindTexture("gui/upgrade/upgrade.png");
+        drawTexturedModalRect(paramInt1, paramInt2, 0, 0, this.xSize, this.ySize);
     }
 
     @Override
     public void drawFG(int paramInt1, int paramInt2, int paramInt3, int paramInt4) {
 // Do nothing because of not implemented
     }
+
+    @Override
+    public void initGui() {
+        int Forward = tileEntity.getForward().getIndex();
+        int Back = tileEntity.getForward().getOpposite().getIndex();
+        int Left = 4;
+        int Right = 5;
+        if (Forward == 3) {
+            Left = 5;
+            Right = 4;
+        }
+        if (Forward == 4) {
+            Left = 3;
+            Right = 2;
+        }
+        if (Forward == 5) {
+            Left = 2;
+            Right = 3;
+        }
+        this.mc.thePlayer.openContainer = this.inventorySlots;
+        this.guiLeft = (this.width - this.xSize) / 2;
+        this.guiTop = (this.height - this.ySize) / 2;
+        this.addButton(new ButtonSides(0, guiLeft + 123, guiTop + 44, sides, guiLeft + 124, guiTop + 45));
+        this.addButton(new ButtonSides(1, guiLeft + 123, guiTop + 20, sides, guiLeft + 124, guiTop + 21));
+        this.addButton(new ButtonSides(Forward, guiLeft + 123, guiTop + 32, sides, guiLeft + 124, guiTop + 33));
+        this.addButton(new ButtonSides(Left, guiLeft + 135, guiTop + 32, sides, guiLeft + 136, guiTop + 33));
+        this.addButton(new ButtonSides(Right, guiLeft + 111, guiTop + 32, sides, guiLeft + 112, guiTop + 33));
+        this.addButton(new ButtonSides(Back, guiLeft + 123, guiTop + 56, sides, guiLeft + 124, guiTop + 57));
+
+    }
+
+    @Override
+    protected void actionPerformed(GuiButton button) throws IOException {
+        //adds side
+        task(button.id);
+    }
+
+    private void task(int i) throws IOException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "task" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AUnusedPrivateMethod)